### DISCUSSION TITLE
Transfering data in Shape layer to AnnData.obs

### DISCUSSION THREAD

**Discussion by tawaunl** [scverse/spatialdata#448] [Q&A] [answered]
In a previous version of Squidpy, there was an option to save data from the shapes layer back to the obs layer of the attached AnnData as a boolean to indicate areas within the shape and areas outside the shape when working with Visium and Xenium data. Do you happen to know if this is still possible? Or is there an easy way to accomplish this

**Comment by LucaMarconato** [comment, marked as answer]
Hi @tawaunl, you can achieve this using the `polygon_query()` function, which is available in the latest version. It will return shapes and the table whose rows are contained inside the polygon used for the query.

The function had a bug leading to poor performance when working with different coordinate system and large raster data, or when querying by multiple polygons; if you experience such problems [please use this PR](https://github.com/scverse/spatialdata/pull/422) with `gh pr checkout 422`.

Please let me know if it works for you.

EDIT: I have added an example on this in the documentation now: https://spatialdata.scverse.org/en/latest/tutorials/notebooks/notebooks/examples/napari_rois.html.

**Reply by tawaunl** [reply]
Thank you! This PR worked out for me. 

One thing to note is that it doesn't add a boolean back to the AnnData table but is very simple to do once we get the query done.

**Comment by bellenger-l** [comment]
Hello @LucaMarconato , 

I'm working with Xenium data and I have spatialdata v0.3.0. I tried the tutorial you mentioned but the format from `polygon_query` is not the same as your example. I don't have the tables. 


```
filtered_tables = {}
for i in range(len(sdata.shapes["tissue_outline"])):
    polygon = sdata["tissue_outline"].geometry.iloc[i]
    filtered_tables[i] = spd.polygon_query(sdata, polygon=polygon, target_coordinate_system="global")

filtered_tables
```

I obtain : 

```
{0: SpatialData object
 ├── Images
 │     ├── 'he_image': DataArray[cyx] (3, 2222, 1772)
 │     ├── 'morphology_focus': DataTree[cyx] (1, 2286, 1416), (1, 1143, 708), (1, 572, 355), (1, 285, 177), (1, 143, 89)
 │     └── 'morphology_mip': DataTree[cyx] (1, 2286, 1416), (1, 1143, 708), (1, 572, 355), (1, 285, 177), (1, 143, 89)
 └── Labels
       ├── 'cell_labels': DataTree[yx] (2286, 1416), (1143, 708), (572, 355), (285, 177), (143, 89)
       └── 'nucleus_labels': DataTree[yx] (2286, 1416), (1143, 708), (572, 355), (285, 177), (143, 89)
 with coordinate systems:
     ▸ 'global', with elements:
         he_image (Images), morphology_focus (Images), morphology_mip (Images), cell_labels (Labels), nucleus_labels (Labels),
 1: SpatialData object
 ├── Images
 │     ├── 'he_image': DataArray[cyx] (3, 3042, 2277)
 │     ├── 'morphology_focus': DataTree[cyx] (1, 2406, 1943), (1, 1203, 972), (1, 601, 486), (1, 301, 243), (1, 151, 122)
 │     └── 'morphology_mip': DataTree[cyx] (1, 2406, 1943), (1, 1203, 972), (1, 601, 486), (1, 301, 243), (1, 151, 122)
 └── Labels
       ├── 'cell_labels': DataTree[yx] (2406, 1943), (1203, 972), (601, 486), (301, 243), (151, 122)
       └── 'nucleus_labels': DataTree[yx] (2406, 1943), (1203, 972), (601, 486), (301, 243), (151, 122)
 with coordinate systems:
     ▸ 'global', with elements:
         he_image (Images), morphology_focus (Images), morphology_mip (Images), cell_labels (Labels), nucleus_labels (Labels),
...
```

So I haven't succeed to finish your tutorial and I am quite stuck...

Can you help me with that one ? 
Best regards,
Lea
